gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -49,7 +49,7 @@
  */
 public class SourceState extends State {
 
-  private static final Set<Extract> extractSet = Sets.newConcurrentHashSet();
+  private static final Set<Extract> EXTRACT_SET = Sets.newConcurrentHashSet();
```

Comment:
`SourceState.createExtract()` invokes a deprecated constructor of `Extract` which adds properties in `SourceState` to the created `Extract`. `ExtractFactory` doesn't do this. Changing this may break existing jobs.